src/rhsm/connection.py

```diff
@@ -127,9 +126,124 @@ def __str__(self):
         return "Bad certificate at %s" % self.cert_path
 
 
+class KeycloakConnection(object):
+    """
+      Keycloak Based Authentication
+    """
+
+    def __init__(self, realm, auth_url, resource, host=None,
+            ssl_port=None,
+            handler=None,
+            proxy_hostname=None,
+            proxy_port=None,
+            proxy_user=None,
+            proxy_password=None,
+            username=None, password=None,
+            cert_file=None, key_file=None,
+            insecure=None,
+            timeout=None,
+            restlib_class=None,
+            correlation_id=None,
+            no_proxy=None,
+            token=None):
```

Comment:
Please try to be consistent in formatting of arguments in this `__init__()` method. Follow PEP8 recommendation.